src/OpenTelemetry/Trace/SpanSdk.cs

```diff
@@ -827,7 +828,24 @@ private bool IsAttributeValueTypeSupported(object attributeValue)
                 return true;
             }
 
-            // TODO add array support
+            if (attributeValue is IEnumerable enumerable)
```

Comment:
this call will result in re-iteration of `IEnumerable`. Perhaps there should be a comment on AddAttribute that will clarify the the IEnumerable will be iterated over multiple times.

Reply:
Added note to `TelemetrySpan.SetAttribyte` to indicate the value may be an enumeration of primitive types and it may be iterated multiple times.